### DISCUSSION TITLE
What should i do if i want to add data augmentation in Yolov8 ?

### DISCUSSION THREAD

**Discussion by ajithkumarmcw** [ultralytics/ultralytics#1950] [Q&A] [answered]
I came through a comment https://github.com/ultralytics/yolov5/issues/2611#issuecomment-1497672844 for data augmentation in yolov5.

Is this right way of doing data augmentation in yolov8 ?

```
results = model.train(data="./data.yaml", epochs=180, imgsz=320,patience=0,
                    project="./runs", name="testrun")  # train the model

```

Adding below content to data.yaml file which is being fed while training
data.yaml is below
```
train: /path/to/train/images
val: /path/to/val/images

nc: 80
names: ['class1', 'class2', 'class3', ...]

# Data augmentation
train_transforms:
  - RandomFlip()
  - RandomRotate(degrees=10)
  - RandomContrast(0.5, 1.5)
  - RandomBrightness(0.5, 1.5)
  - RandomSaturation(0.5, 1.5)
  - Resize((640, 640), pad_mode='letterbox')
  - ToTensor()

# Validation transforms
val_transforms:
  - Resize((640, 640), pad_mode='letterbox')
  - ToTensor()
```

**Comment by glenn-jocher** [comment, marked as answer]
@ajithkumarmcw sorry, that's not a correct usage. You can modify the hyps directly in default.yaml or pass them as arguments to a Python or CLI training command. For details see https://docs.ultralytics.com/modes/train/

**Reply by M-Amrollahi** [reply]
When I try to train a Yolov8 model, I get these logs among training:
```
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
```

So, how can I change them?

**Reply by ajithkumarmcw** [reply]
@M-Amrollahi  you can modify it in https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/default.yaml or if its not available there u can modify here directly https://github.com/ultralytics/ultralytics/blob/main/ultralytics/data/augment.py#L767-L795

**Reply by M-Amrollahi** [reply]
Do you know, how I can modify the code for validation dataset?

**Reply by ajithkumarmcw** [reply]
i think this https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/default.yaml#L66 not sure

**Reply by M-Amrollahi** [reply]
I have tried changing ```Albumentation``` from the source code, so I just wanted to apply it through the source but noy yaml file.
I do not know where I should apply transformation for validation dataset.

**Reply by ajithkumarmcw** [reply]
I too searched it i am also not sure of it , May be if u raise a new discussion or issue u might get help

**Reply by MilenioScience** [reply]
I still confuse how to do data augmentation, okay so assume that i have default.yaml file, and also i have my data.yaml from my custom dataset, is it true that i have to set the "data" parameter to default.yaml instead of data.yaml in model.train command?

**Reply by MilenioScience** [reply]
Or what parameter i need to set to pass the default.yaml in model.train command?

**Reply by pderrenger** [reply]
@MilenioScience to apply data augmentations during training with YOLOv8, you should modify the hyperparameter (hyps) settings, which are specified in the `default.yaml` file. You do not need to pass the `default.yaml` file directly to the `model.train()` command. Instead, you can either:

1. Directly edit the `default.yaml` file to include your desired augmentation settings under the appropriate keys (e.g., `hsv_h`, `hsv_s`, `hsv_v`, `degrees`, `translate`, etc.).
2. Create a custom hyperparameter file (e.g., `custom_hyps.yaml`) with your desired augmentations and pass the path to this file using the `cfg` parameter in the `model.train()` command.

Here's an example of how you might use a custom hyperparameter file:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Train the model with custom hyperparameters
results = model.train(data='data.yaml', cfg='custom_hyps.yaml', epochs=3)
```

In your `data.yaml`, you only need to specify the paths to your training and validation datasets, the number of classes, and class names. The augmentation settings should be in the hyperparameter file.

Remember that the `data` parameter in the `model.train()` command should always point to your dataset configuration file (e.g., `data.yaml`), which includes the paths to your training and validation data. The `cfg` parameter is used to specify a custom hyperparameter configuration if you're not using the default settings.
